File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
from collections import deque

def solution(number, limit, power):
    divisor_lst = [0] * (number + 1)
    
    for i in range(1, number+1):
        for j in range(i, number+1, i):
            divisor_lst[j] += 1
                    
    my_queue = deque(divisor_lst)
    result = 0
    
    while my_queue:
        cur_divisor = my_queue.popleft()
        if cur_divisor > limit:
            result += power
        else:
            result += cur_divisor
    
    return result
            
        

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
const solution = (number, limit, power) => {
    let result = 0;
    const divisor_lst = Array(number + 1).fill(0); 
    for (let i=1; i<number+1; i++) { 
        for (let j=i; j<number+1; j+=i) { 
            divisor_lst[j] += 1;
        }
    }
    divisor_lst.forEach((num) => {
        if (num <= limit) {
            result += num;
        } else {
            result += power;
        }
    });
    return result;
}